tensorflow/tensorboard
# Add "Upload" button to header for local users

* Motivation for features / changes

  Show the "Upload" button to users who load a TensorBoard on localhost. This Upload button is an entry point to the "Upload to TensorBoard.dev" dialog with instructions on how to create a TensorBoard.dev experiment with the event data.  We restrict to localhost users so we can be more certain that the user is the owner of the TensorBoard.

* Technical description of changes

  In the header, check if window.location.hostname is a "localhost" or "127.0.0.1" value.  If so, show the button.

  Also, improve the styling of the button.

  Also, remove "?" from the "Upload to TensorBoard" dialog.

* Screenshots of UI changes

  On localhost:8080 or 127.0.0.1:
  ![image](https://user-images.githubusercontent.com/17152369/86126795-e78adc80-baac-11ea-962a-089bd830c6d6.png)

  On non-localhost:
  ![image](https://user-images.githubusercontent.com/17152369/86126914-1bfe9880-baad-11ea-85b5-9911a1a89749.png)

* Detailed steps to verify changes work correctly (as executed by you)
  * Write some tests.
  * Manually verify:
    * Edit hosts file so www.nytimes.com points to localhost.
    * Load ng_index.html using localhost:<port>, 127.0.0.1:<port>, and www.nytimes.com:<port>
    * Observe in the the first two instances that the "Upload" button appears, with new styling.
    * Observce in the final instance that the "Upload" button does not appear.
